ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -2,199 +2,238 @@
-    
-    def _save_doc(self,doc_id:str) -> Tuple[str,dict]:
-        """Save document pages as json and return filename along with document's metadata"""
-
-        # get metadata and pages for this doc
-        meta,pages = self._get_meta_pages(doc_id)
-
-        # if pages, save json
-        if pages:
-            filename = os.path.join(self.path_texts, meta[self.WORK_ID_FIELD]+'.json')
-            os.makedirs(self.path_texts,exist_ok=True)
-            with open(filename,'w') as of:
-                json.dump(pages, of, indent=2)
-        
-        # otherwise, returned filename is blank to indicate no file saved
-        else:
-            filename=''
-        
-        return filename,meta
+        # add --doc-limit argument (determines how many results to retrieve from solr)
+        parser.add_argument(
+            "--doc-limit",
+            type=int,
+            default=-1,
+            help="Limit on the number of documents for corpus generation."
+            "The default of -1 considers ALL documents.",
+        )
 
+        # add --batch-size argument (for solr querying)
+        parser.add_argument(
+            "--batch-size",
+            type=int,
+            default=DEFAULT_BATCH_SIZE,
+            help="Number of docs to query from solr at one time",
+        )
 
-    def save(self):
-        """Save the generated corpus text and metadata to files on disk"""
+        # add --dry-run argument (don't save anything, just iterate)
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Do not save anything, just iterate over solr",
+        )
 
-        # save docs and gather metadata
-        metadata=[]
-        pdesc='Saved text to'
-        pbar=tqdm(total=self.doc_count, desc=f'{pdesc}: ...')
-        for doc_id in self.doc_ids:
-            # get saved filename and found metadata for this document
-            fn,meta = self._save_doc(doc_id)
+    #### SOLR ####
 
-            # if we saved, update progress bar desc
-            if fn: pbar.set_description(f'{pdesc}: {fn}')
+    @cached_property
+    def query_set(self):
+        """
+        A cached representation of the query set;
+        it discourages opening multiple connections to database.
+        """
+        return SolrQuerySet()
 
-            # tick
-            pbar.update()
+    def iter_solr(self, item_type="page"):
+        """
+        Iterate over solr documents of a certain `item_type`.
+        """
 
-            # add this doc's meta to metadata
-            metadata.append(meta)
-        pbar.close()
+        # get params
+        batch_size, lim = self.batch_size, self.doclimit
 
-        # save metadata csv
-        with open(self.path_metadata,'w') as of:
-            json.dump(metadata, of, indent=2)
-        print(f'Saved metadata to: {self.path_metadata}')
+        # search for this item type, order by id
+        qset = self.query_set.search(item_type=item_type).order_by("id")
```

Comment:
I think `filter` might be slightly better than `search` based on how Solr caches things, although it might not make a difference here.

```suggestion
        qset = self.query_set.filter(item_type=item_type).order_by("id")
```